docs/dev_guide/middleware.md

```diff
@@ -0,0 +1,71 @@
+Sometimes you need to implement a middleware that has to be executed on every call. For instance if
+ you want to authenticate in differents ways depending on the path or force HTTPS and WSS. For this
+ reason you could use middlewares that follows the 
+  [ASGI](https://asgi.readthedocs.io/en/latest/specs/main.html) protocol standards.
+  [FastAPI](https://fastapi.tiangolo.com/advanced/middleware/) and
+   [Starlette](https://www.starlette.io/middleware/) have already implemented some of them.
+
+Basically you could write your own middleware in a two different ways:
```

Comment:
I think it's better to redirect the user to the starlette/fastapi docs to understand how they can implement a middleware. Our goal is to explain that, if they want to configure custom middlewares, they have to override the get_middlewares base function.

Reply:
Do you mean deleting the reference to ASGI? But we also need to explain how to implement the middleware, don't we?